crates/sui-indexer/src/store/pg_indexer_store.rs

```diff
@@ -144,208 +129,229 @@ impl PgIndexerStore {
-                .first::<Vec<u8>>(conn)
-                .optional()
-        })
-        .context("Failed reading chain id from PostgresDB")
+    async fn get_chain_identifier(&self) -> Result<Option<Vec<u8>>, IndexerError> {
+        use diesel_async::RunQueryDsl;
+
+        let mut connection = self.pool.get().await?;
+
+        chain_identifier::table
+            .select(chain_identifier::checkpoint_digest)
+            .first::<Vec<u8>>(&mut connection)
+            .await
+            .optional()
+            .map_err(Into::into)
+            .context("Failed reading chain id from PostgresDB")
     }
 
-    fn get_latest_checkpoint_sequence_number(&self) -> Result<Option<u64>, IndexerError> {
-        read_only_blocking!(&self.blocking_cp, |conn| {
-            checkpoints::dsl::checkpoints
-                .select(max(checkpoints::sequence_number))
-                .first::<Option<i64>>(conn)
-                .map(|v| v.map(|v| v as u64))
-        })
-        .context("Failed reading latest checkpoint sequence number from PostgresDB")
+    async fn get_latest_checkpoint_sequence_number(&self) -> Result<Option<u64>, IndexerError> {
+        use diesel_async::RunQueryDsl;
+
+        let mut connection = self.pool.get().await?;
+
+        checkpoints::table
+            .select(max(checkpoints::sequence_number))
+            .first::<Option<i64>>(&mut connection)
+            .await
+            .map_err(Into::into)
+            .map(|v| v.map(|v| v as u64))
+            .context("Failed reading latest checkpoint sequence number from PostgresDB")
     }
 
-    fn get_available_checkpoint_range(&self) -> Result<(u64, u64), IndexerError> {
-        read_only_blocking!(&self.blocking_cp, |conn| {
-            checkpoints::dsl::checkpoints
-                .select((
-                    min(checkpoints::sequence_number),
-                    max(checkpoints::sequence_number),
-                ))
-                .first::<(Option<i64>, Option<i64>)>(conn)
-                .map(|(min, max)| {
-                    (
-                        min.unwrap_or_default() as u64,
-                        max.unwrap_or_default() as u64,
-                    )
-                })
-        })
-        .context("Failed reading min and max checkpoint sequence numbers from PostgresDB")
+    async fn get_available_checkpoint_range(&self) -> Result<(u64, u64), IndexerError> {
+        use diesel_async::RunQueryDsl;
+
+        let mut connection = self.pool.get().await?;
+
+        checkpoints::table
+            .select((
+                min(checkpoints::sequence_number),
+                max(checkpoints::sequence_number),
+            ))
+            .first::<(Option<i64>, Option<i64>)>(&mut connection)
+            .await
+            .map_err(Into::into)
+            .map(|(min, max)| {
+                (
+                    min.unwrap_or_default() as u64,
+                    max.unwrap_or_default() as u64,
```

Comment:
not related to this PR just flagging that it doesn't seem ideal for us to return a default value for the checkpoint range if we can't actually retrieve these from the table

Reply:
Yeah I noticed a few of these that we should try and fix at some point